plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Data.hs

```diff
@@ -0,0 +1,206 @@
+      , \n -> Map.fromList $
+          [ ("type:", noopCompleter)
+          , ("main-is:", filePathCompleter)
+          , ("other-modules:", modulesCompleter (sourceDirsExtractionBenchmark n))
+          ]
+            ++ libExecTestBenchCommons
+      )
+    ,
+      ( "foreign-library"
+      , \_ -> Map.fromList
+          [ ("type:", constantCompleter ["native-static", "native-shared"])
+          , ("options:", constantCompleter ["standalone"])
+          , ("mod-def-file:", filePathCompleter)
+          , ("lib-version-info:", noopCompleter)
+          , ("lib-version-linux:", noopCompleter)
+          ]
+      )
+    ,
+      ( "flag"
+      , \_ -> Map.fromList
+          [ ("description:", noopCompleter)
+          , ("default:", constantCompleter ["True", "False"])
+          , ("manual:", constantCompleter ["False", "True"])
+          , ("lib-def-file:", noopCompleter)
+          , ("lib-version-info:", noopCompleter)
+          , ("lib-version-linux:", noopCompleter)
+          ]
+      )
+    ,
+      ( "source-repository"
+      , \_ -> Map.fromList
+          [
+            ( "type:"
+            , constantCompleter
+                [ "darcs"
+                , "git"
+                , "svn"
+                , "cvs"
+                , "mercurial"
+                , "hg"
+                , "bazaar"
+                , "bzr"
+                , "arch"
+                , "monotone"
+                ]
+            )
+          , ("location:", noopCompleter)
+          , ("module:", noopCompleter)
+          , ("branch:", noopCompleter)
+          , ("tag:", noopCompleter)
+          , ("subdir:", directoryCompleter)
+          ]
+      )
+    ]
+ where
+  libExecTestBenchCommons =
+    [ ("build-depends:", noopCompleter)
+    , ("hs-source-dirs:", directoryCompleter)
+    , ("default-extensions:", noopCompleter)
+    , ("other-extensions:", noopCompleter)
+    , ("default-language:", constantCompleter ["GHC2021", "Haskell2010", "Haskell98"])
+    , ("other-languages:", noopCompleter)
+    , ("build-tool-depends:", noopCompleter)
+    , ("buildable:", constantCompleter ["True", "False"])
+    , ("ghc-options:", noopCompleter) -- todo: maybe there is a list of possible ghc options somewhere
+    , ("ghc-prof-options:", noopCompleter)
+    , ("ghc-shared-options:", noopCompleter)
+    , ("ghcjs-options:", noopCompleter)
+    , ("ghcjs-prof-options:", noopCompleter)
+    , ("ghcjs-shared-options:", noopCompleter)
+    , ("includes:", filePathCompleter) -- list of filenames
+    , ("install-includes:", filePathCompleter) -- list of filenames
+    , ("include-dirs:", directoryCompleter) -- list of directories
+    , ("c-sources:", filePathCompleter) -- list of filenames
+    , ("cxx-sources:", filePathCompleter) -- list of filenames
+    , ("asm-sources:", filePathCompleter) -- list of filenames
+    , ("cmm-sources:", filePathCompleter) -- list of filenames
+    , ("js-sources:", filePathCompleter) -- list of filenames
```

Comment:
The comments seem a bit superfluous.

Reply:
Yes, they are my own documentation from when i noted down where i could possibly suggest completions other than noopCompleter. I will remove them.